### PR TITLE
Check validity of symbols in Ycheck

### PR DESCRIPTION
**Pull request by DarkDimius**
based on #377

### CODE REVIEW

**Comment by DarkDimius** [comment]
@odersky please review. Also this change makes easy to spot #371

File: src/dotty/tools/dotc/transform/TreeChecker.scala [hunk 1]
```diff
@@ -35,10 +35,47 @@ import scala.util.control.NonFatal
...
+
+  def printError(str: String) = {
+    println(Console.RED + "[error] " + Console.WHITE  + str)
+  }
+
+  val NoSuperClass = Trait | Package
+
+  def testDuplicate(sym: Symbol, registry: mutable.Map[String, Symbol], typ: String)(implicit ctx: Context) = {
+    val name = sym.fullName.toString
+    if(registry.contains(name))
```
Comment (odersky) [inline src/dotty/tools/dotc/transform/TreeChecker.scala]: Formatting: Leave a space between if and (.

**Comment by odersky** [comment]
Otherwise LGTM

File: src/dotty/tools/dotc/transform/TreeChecker.scala [hunk 2]
```diff
@@ -35,10 +35,47 @@ import scala.util.control.NonFatal
  *   - After typer, identifiers and select nodes refer to terms only (all types should be
  *     represented as TypeTrees then).
  */
-class TreeChecker extends Phase {
+class TreeChecker extends Phase with SymTransformer {
   import ast.tpd._
 
 
+  val seenClasses = collection.mutable.HashMap[String, Symbol]()
+  val seenModuleVals = collection.mutable.HashMap[String, Symbol]()
```
Comment (retronym) [inline src/dotty/tools/dotc/transform/TreeChecker.scala]: Seems like these ought to be private.

**Pull request merged**